build.zig

```diff
@@ -0,0 +1,383 @@
+        "dbg_mlc.c",
+        "dyn_load.c",
+        "finalize.c",
+        "headers.c",
+        "mach_dep.c",
+        "malloc.c",
+        "mallocx.c",
+        "mark.c",
+        "mark_rts.c",
+        "misc.c",
+        "new_hblk.c",
+        "obj_map.c",
+        "os_dep.c",
+        "ptr_chck.c",
+        "reclaim.c",
+        "typd_mlc.c",
+    }) catch unreachable;
+
+    if (t.abi.isMusl()) {
+        std.log.debug("Musl detected", .{});
+        lib.defineCMacro("NO_GETCONTEXT", "1");
+    }
+
+    // Always enabled
+    lib.defineCMacro("ALL_INTERIOR_POINTERS", "1");
+    lib.defineCMacro("NO_EXECUTE_PERMISSION", "1");
+    // Zig comes with clang that supports atomics so we can hardcode this
+    lib.defineCMacro("GC_BUILTIN_ATOMIC", "1");
+    // TODO: Should this be configurable?
+//    lib.defineCMacro("NO_PROC_FOR_LIBRARIES", "1");
+    flags.appendSlice(&.{
+        "-fno-sanitize=undefined",
+    }) catch unreachable;
+
+    if (enable_threads) {
+        source_files.appendSlice(&.{
+            "gc_dlopen.c",
+            "pthread_start.c",
+            "pthread_support.c"
+        }) catch unreachable;
+        lib.defineCMacro("GC_THREADS", "1");
+        lib.defineCMacro("_REENTRANT", "1");
+        if (enable_parallel_mark) {
+            lib.defineCMacro("PARALLEL_MARK", "1");
+        }
+        if (enable_thread_local_alloc) {
+            lib.defineCMacro("THREAD_LOCAL_ALLOC", "1");
+            source_files.appendSlice(&.{
+                "specific.c",
+                "thread_local_alloc.c",
+            }) catch unreachable;
+        }
+        if (t.os.tag == .windows) {
+            source_files.appendSlice(&.{
+                "win32_threads.c"
+            }) catch unreachable;
+        } else {
+            if (t.os.tag == .macos) {
+                source_files.appendSlice(&.{
+                    "darwin_stop_world.c"
+                }) catch unreachable;
+            } else {
+                source_files.appendSlice(&.{
+                    "pthread_stop_world.c"
+                }) catch unreachable;
+            }
+        }
+    }
+
+    if (disable_handle_fork) {
+        lib.defineCMacro("NO_HANDLE_FORK", "1");
+    }
+
+    if (enable_gcj_support) {
+        lib.defineCMacro("GC_GCJ_SUPPORT", "1");
+        source_files.appendSlice(&.{
+            "gcj_mlc.c"
+        }) catch unreachable;
+    }
+
+    if (enable_disclaim) {
+        lib.defineCMacro("ENABLE_DISCLAIM", "1");
+        source_files.appendSlice(&.{
+            "fnlz_mlc.c"
+        }) catch unreachable;
+    }
+
+    if (enable_dynamic_pointer_mask) {
+        lib.defineCMacro("DYNAMIC_POINTER_MASK", "1");
+    }
+
+    if (enable_java_finalization) {
+        lib.defineCMacro("JAVA_FINALIZATION", "1");
+    }
+
+    if (enable_atomic_uncollectable) {
+        lib.defineCMacro("GC_ATOMIC_UNCOLLECTABLE", "1");
+    }
+
+    if (enable_redirect_malloc) {
+        lib.defineCMacro("REDIRECT_MALLOC=GC_malloc", "1");
```

Comment:
I guess it should be: lib.defineCMacro("REDIRECT_MALLOC", "GC_malloc")

Reply:
Yeah that was wrong, but I'm changing back to the use of flags so it's correct again :)